indico/modules/events/registration/client/js/form/fields/DateInput.jsx

```diff
@@ -42,18 +42,14 @@ export function DateInputComponent({
   return (
     <Form.Group styleName="date-field">
       <Form.Field>
-        <SingleDatePicker
+        <Datepicker
           id={id}
           name="date"
           disabled={disabled}
           required={required}
-          date={toMoment(dateValue, 'YYYY-MM-DD', true)}
-          onDateChange={handleDateChange}
-          placeholder={dateFormat}
-          displayFormat={dateFormat}
-          isOutsideRange={() => false}
-          verticalSpacing={10}
-          enableOutsideDays
+          format={dateFormat}
+          value={dateValue}
+          onChange={handleDateChange}
```

Comment:
You also need to pass through the `onFocus` and `onBlur` props from the `DateInputComponent`. That way the final-form field will be correctly marked as "touched" and will immediately show validation errors on blur.